### PR TITLE
move crc32c to stdlib

### PR DESCRIPTION
**Pull request by KristofferC**
Fix https://github.com/JuliaLang/julia/issues/24235

CC @stevengj, @StefanKarpinski

### CODE REVIEW

File: base/deprecated.jl [hunk 1]
```diff
@@ -1354,6 +1354,8 @@ export conv, conv2, deconv, filt, filt!, xcorr
 @deprecate_moved watch_file "FileWatching" true true
 @deprecate_moved FileMonitor "FileWatching" true true
 
+@deprecate_moved crc32c "CRC32C" true true
```
Comment (stevengj) [inline base/deprecated.jl]: Later on you don't capitalize the last letter in "CRC32c"?

**Review by stevengj — APPROVED** [review]

**Review by stevengj — APPROVED** [review]

File: stdlib/CRC32c/test/runtests.jl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    end
+end
+unsafe_crc32c_sw(a, n, crc) =
+    ccall(:jl_crc32c_sw, UInt32, (UInt32, Ptr{UInt8}, Csize_t), crc, a, n)
+crc32c_sw(a::Union{Array{UInt8},Base.FastContiguousSubArray{UInt8,N,<:Array{UInt8}} where N},
+          crc::UInt32=0x00000000) = unsafe_crc32c_sw(a, length(a), crc)
+crc32c_sw(s::String, crc::UInt32=0x00000000) = unsafe_crc32c_sw(s, sizeof(s), crc)
+function crc32c_sw(io::IO, nb::Integer, crc::UInt32=0x00000000)
+    nb < 0 && throw(ArgumentError("number of bytes to checksum must be ≥ 0"))
+    buf = Array{UInt8}(min(nb, 24576))
```
Comment (musm) [inline stdlib/CRC32c/test/runtests.jl]: Unrelated but maybe you can change this to a `Vector`

Comment (stevengj) [reply stdlib/CRC32c/test/runtests.jl]: It is a vector already.  Not sure why you object to the `Array{T}(n)` constructor, which is documented.

Comment (musm) [reply stdlib/CRC32c/test/runtests.jl]: I thought it should  help the  reduce runtime cost if we explicitly specify it as a one-dimensional array

Comment (yuyichao) [reply stdlib/CRC32c/test/runtests.jl]: No it does not reduce any runtime cost. It's only recommended for clarity and minor type inference performance.

**Review by StefanKarpinski — APPROVED** [review]
LGTM!

**Pull request merged**